nntrainer/layers/lstmcell_core.cpp

```diff
@@ -0,0 +1,519 @@
+  recurrent_acti_func.setActiFunc(recurrent_activation_type.get());
+}
+
+void LSTMCellCoreLayer::setProperty(const std::vector<std::string> &values) {
+  std::vector<std::string> remain_props =
+    loadProperties(values, lstmcell_core_props);
+  LayerImpl::setProperty(remain_props);
+}
+
+void LSTMCellCoreLayer::exportTo(Exporter &exporter,
+                                 const ExportMethods &method) const {
+  LayerImpl::exportTo(exporter, method);
+  exporter.saveResult(lstmcell_core_props, method, this);
+}
+
+void LSTMCellCoreLayer::forwarding(RunLayerContext &context, bool training) {
+  const unsigned int unit = std::get<props::Unit>(lstmcell_core_props).get();
+
+  const Tensor &input = context.getInput(INDEX::INPUT);
+  const Tensor &prev_hidden_state = context.getInput(INDEX::HIDDEN_STATE_IN);
+  const Tensor &prev_cell_state = context.getInput(INDEX::CELL_STATE_IN);
+  const TensorDim &input_dim = input.getDim();
+  const unsigned int batch_size = input_dim.batch();
+
+  Tensor &next_hidden_state = context.getOutput(INDEX::HIDDEN_STATE_OUT);
+  Tensor &next_cell_state = context.getOutput(INDEX::CELL_STATE_OUT);
+
+#if (ENABLE_SHARING_WT_IDX)
+  const Tensor &weight_ih =
+    context.getWeight(wt_idx[LSTMCellCoreParams::weight_ih]);
+  const Tensor &weight_hh =
+    context.getWeight(wt_idx[LSTMCellCoreParams::weight_hh]);
+  const Tensor &bias_ih =
+    context.getWeight(wt_idx[LSTMCellCoreParams::bias_ih]);
+#else
+  const Tensor &weight_ih = context.getWeight(LSTMCellCoreParams::weight_ih);
+  const Tensor &weight_hh = context.getWeight(LSTMCellCoreParams::weight_hh);
+  const Tensor &bias_ih = context.getWeight(LSTMCellCoreParams::bias_ih);
+#endif
+
+#if ENABLE_SHARING_WT_IDX
+  Tensor &ifgo = context.getTensor(wt_idx[LSTMCellCoreParams::ifgo]);
+#else
+  Tensor &ifgo = context.getTensor(0);
+#endif
+
+  input.dot(weight_ih, ifgo);
+  prev_hidden_state.dot(weight_hh, ifgo, false, false, 1.0);
+  ifgo.add_i(bias_ih);
+
+  Tensor input_forget_gate =
+    ifgo.getSharedDataTensor({batch_size, 1, 1, unit * 2}, 0, false);
+  Tensor input_gate =
+    ifgo.getSharedDataTensor({batch_size, 1, 1, unit}, 0, false);
+  Tensor forget_gate =
+    ifgo.getSharedDataTensor({batch_size, 1, 1, unit}, unit, false);
+  Tensor memory_cell =
+    ifgo.getSharedDataTensor({batch_size, 1, 1, unit}, unit * 2, false);
+  Tensor output_gate =
+    ifgo.getSharedDataTensor({batch_size, 1, 1, unit}, unit * 3, false);
+
+  recurrent_acti_func.run_fn(input_forget_gate, input_forget_gate);
+  recurrent_acti_func.run_fn(output_gate, output_gate);
+  acti_func.run_fn(memory_cell, memory_cell);
+
+  forget_gate.multiply_strided(prev_cell_state, next_cell_state);
+  memory_cell.multiply_strided(input_gate, next_cell_state, 1.0f);
+
+  acti_func.run_fn(next_cell_state, next_hidden_state);
+  next_hidden_state.multiply_i_strided(output_gate);
+}
+
+void LSTMCellCoreLayer::calcDerivative(RunLayerContext &context) {}
```

Comment:
for now this is ok, but later this should do derivative calculation for the LSTMCellCore to be part of a model training as non-trainable layer.